### DISCUSSION TITLE
How to optimally change 1 parameter in slint model?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#4595] [Q&A]
I have model that usually is used in even million of items
```
export struct MainListModel {
    checked: bool,
    header_row: bool,
    selected_row: bool,
    val: [string]
}
```
which I load into 
```
let model ModelRc<MainListModel> = app.get_model();
```
I tested three possible solutions, but only first not require to copy/clone data, but it not works
```
fn select_all(model: ModelRc<MainListModel>, app: &MainWindow) {
// Not works
    for mut item in model.iter() {
        item.checked = true;
    }
    dbg!(&model.iter().nth(0).unwrap().checked);

// Works
    let mut new_data = model.iter().nth(1).unwrap().clone();
    new_data.checked = true;
    model.set_row_data(1, new_data);
    dbg!(&model.iter().nth(1).unwrap().checked);

// Works
    let mut old_data = model.iter().collect::<Vec<_>>();
    old_data.iter_mut().for_each(|x| x.checked = true);
    let new_model = ModelRc::new(VecModel::from(old_data));
    app.set_empty_files_model(new_model);
    dbg!(&new_model.iter().nth(2).unwrap().checked);
}

```

**Comment by ogoffart** [comment]
I'd say your second option is the most optimal right now.

But it still invalidates the entire struct. We should so something in Slint to support this usecase, for example, we could generating one property per struct fields in the generated code.

Right now, if this causes performances problem, you can also try to split the MainListModel struct and use several model: put the "checked" field ouside of the MainListModel, so you have another `in property <[bool]> checked` that you access with `checked[the_index]`.
Not the most convenient, but should make invalidation less expensive.
